modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/parse/MBObjectParser.java

```diff
@@ -639,8 +639,8 @@ public Expression percentage(JSONObject json, String tag, Number fallback)
             throw new MBFormatException("\"" + tag + "\" percentage from boolean " + obj
                     + " not supported, expected value between 0 and 1");
         } else if (obj instanceof JSONObject) {
-            throw new UnsupportedOperationException(
-                    "\"" + tag + "\"precentage from Function not yet supported");
+            MBFunction function = new MBFunction(this, (JSONObject) obj);
+            return function.numeric();
```

Comment:
In the mapbox spec for these cases, the parameter value is generally of type 'number' and are supposed to be between 0 and 1. Unless we want to be extremely forgiving, I think 50% should fail (and it currently would).